### PR TITLE
fix not showing base error bug

### LINKED ISSUE #19339: Not showing error in options flow window
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

When I try to submit not valid options in Configure window for integration I got error, but error doesnt shown in UI
options
But backend returns the error
```
{
    "type": "form",
...
    "errors": {
        "base": "test error"
    },
    "description_placeholders": null,
    "last_step": null,
    "preview": null
}
```

### Describe the behavior you expected

It should work like in the integration setup window
setup
Backend returns same errors
```
{
    "type": "form",
...
    "errors": {
        "base": "test error"
    },
    "description_placeholders": null,
    "last_step": null,
    "preview": null
}
```

### Steps to reproduce the issue

Select any custom integration with OptionsFlow in config_flow.py
And in return self.async_show_form(...) add errors like
```
return self.async_show_form(step_id="init", data_schema=data_schema, errors={"base": "test error"})
```

### What version of Home Assistant Core has the issue?

2024.1.2

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

_No response_

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by AlexandrErohin**
## Proposed change
There is a bug for not showing base error in options flow window. https://github.com/home-assistant/frontend/issues/19339
When an error is not existed in translations, it becomes empty. It was fixed in config flow wondow https://github.com/home-assistant/frontend/blob/dev/src/dialogs/config-flow/show-dialog-config-flow.ts#L100

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #19339
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/dialogs/config-flow/show-dialog-options-flow.ts [hunk 1]
```diff
@@ -110,9 +110,11 @@ export const showOptionsFlowDialog = (
...
       renderShowFormStepFieldError(hass, step, error) {
-        return hass.localize(
-          `component.${configEntry.domain}.options.error.${error}`,
-          step.description_placeholders
+        return (
+            hass.localize(
+                `component.${configEntry.domain}.options.error.${error}`,
+                step.description_placeholders
+            ) || error
         );
```
Comment (bramkragten) [inline src/dialogs/config-flow/show-dialog-options-flow.ts:118]:
```suggestion
        return
            hass.localize(
                `component.${configEntry.domain}.options.error.${error}`,
                step.description_placeholders
        ) || error;
```

File: src/dialogs/config-flow/show-dialog-options-flow.ts [hunk 2]
```diff
@@ -110,9 +110,11 @@ export const showOptionsFlowDialog = (
...
 
       renderShowFormStepFieldError(hass, step, error) {
-        return hass.localize(
-          `component.${configEntry.domain}.options.error.${error}`,
-          step.description_placeholders
+        return (
+            hass.localize(
+                `component.${configEntry.domain}.options.error.${error}`,
+                step.description_placeholders
+            ) || error
```
Comment (karwosts) [inline src/dialogs/config-flow/show-dialog-options-flow.ts:117]:
```suggestion
        return (
          hass.localize(
            `component.${configEntry.domain}.options.error.${error}`,
            step.description_placeholders
          ) || error
```

Whitespace needs to be exact to satisfy the lint check, it's somewhat tricky if you don't have a development environment setup.

**Comment by AlexandrErohin** [comment]
@bramkragten Thanks!

**Comment by AlexandrErohin** [comment]
Is there anything I need to do next?

**Review by bramkragten — APPROVED** [review]

**Pull request merged**